components/style/values/generics/grid.rs

```diff
@@ -556,3 +557,93 @@ impl<T: ToCss> ToCss for TrackList<T> {
         Ok(())
     }
 }
+
+/// The `<line-name-list>` for subgrids.
+///
+/// `[ <line-names> | repeat(<positive-integer> | auto-fill, <line-names>+) ]+`
+/// **NOTE; This is deprecated!**
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub struct LineNameList {
+    /// The optional `<line-name-list>`
+    pub names: Vec<Vec<CustomIdent>>,
```

Comment:
I wonder if that should be `Box<[Box<[CustomIdent]>>`, but that's unrelated to that patch.